File: app/src/main/java/com/mkf_test/showtexts/ui/ShowTextActivity2.java
package com.mkf_test.showtexts.ui;

import android.os.Bundle;
import android.text.Html;
import android.view.View;

import com.mkf_test.showtexts.BaseActivity;
import com.mkf_test.showtexts.ParseHttp.ParseHttpListener;
import com.mkf_test.showtexts.R;
import com.mkf_test.showtexts.entity.Book;
import com.mkf_test.showtexts.entity.ParseHttpData;
import com.mkf_test.showtexts.utils.ParseHttpUtils;
import com.mkf_test.showtexts.widget.FlipView;

import org.xutils.view.annotation.ContentView;
import org.xutils.view.annotation.ViewInject;

@ContentView(R.layout.activity_show_text2)
public class ShowTextActivity2 extends BaseActivity  {
//    @ViewInject(R.id.iv_text)
//    ImageView iv_text;
//    @ViewInject(R.id.viewpager)
//ViewPager viewpager;
    @ViewInject(R.id.flip_view)
    FlipView mFlipView;
    String urlPath;
    ParseHttpData data;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setToolbarVisibility(View.GONE);
//        setContentView(R.layout.activity_show_text2);
//        mFlipView = (FlipView) findViewById(R.id.flip_view);
        urlPath= getIntent().getStringExtra("url");
        ParseHttp(0);

        mFlipView.setOnPageFlippedListener(new FlipView.OnPageFlippedListener() {
            @Override
            public boolean onPageLast() {
                if (data.getNextUrl()!=null&&data.getNextUrl().getUrl()!=null)   {
                    urlPath=data.getNextUrl().getUrl();
                    ParseHttp(1);
                }
                return true;
            }

            @Override
            public boolean onPageStart() {
                if (data.getPrevUrl()!=null&&data.getPrevUrl().getUrl()!=null)    {
                    urlPath=data.getPrevUrl().getUrl();
                    ParseHttp(2);
                }
                return true;
            }

            @Override
            public void onFlipStarted() {

            }

            @Override
            public void onFoldViewClicked() {

            }
        });

    }
    private void ParseHttp(final int i) {
        ParseHttpUtils.ParseFromHttp(urlPath, new ParseHttpListener() {
            @Override
            public void ParseHttpSuccess(ParseHttpData data) {
                if (data!=null){
                    getSharedPreferences(getPackageName(),MODE_PRIVATE).edit().putString("url",urlPath).commit();
                    ShowTextActivity2.this.data=data;
                    initViewData(i);
                }
            }
        });
    }
    private void initViewData(int i) {
        if (data.getIsCatalog()==1){
        }else {
            Book book = new Book(data.getTitle()==null?"":data.getTitle(),
                    Html.fromHtml(data.getText()==null?"":data.getText()).toString());
            mFlipView.setBook(book,i);
        }
    }
}

File: app/src/main/java/com/mkf_test/showtexts/ui/ShowTextActivity2.kt
package com.mkf_test.showtexts.ui

import android.os.Bundle
import android.text.Html
import android.view.View
import com.mkf_test.showtexts.BaseActivity
import com.mkf_test.showtexts.R
import com.mkf_test.showtexts.entity.Book
import com.mkf_test.showtexts.entity.ParseHttpData
import com.mkf_test.showtexts.utils.ParseHttpUtils
import com.mkf_test.showtexts.widget.FlipView
import kotlinx.android.synthetic.main.activity_show_text2.*

class ShowTextActivity2 : BaseActivity() {
    internal var urlPath: String? = null
    internal var data: ParseHttpData? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setToolbarVisibility(View.GONE)
        //        setContentView(R.layout.activity_show_text2);
        //        mFlipView = (FlipView) findViewById(R.id.flip_view);
        urlPath = intent.getStringExtra("url")
        ParseHttp(0)

        flip_view.setOnPageFlippedListener(object : FlipView.OnPageFlippedListener {
            override fun onPageLast(): Boolean {
                if (data!!.nextUrl != null && data!!.nextUrl.url != null) {
                    urlPath = data!!.nextUrl.url
                    ParseHttp(1)
                }
                return true
            }

            override fun onPageStart(): Boolean {
                if (data!!.prevUrl != null && data!!.prevUrl.url != null) {
                    urlPath = data!!.prevUrl.url
                    ParseHttp(2)
                }
                return true
            }

            override fun onFlipStarted() {

            }

            override fun onFoldViewClicked() {

            }
        })

    }

    override fun getLayoutResId(): Int {
        return R.layout.activity_show_text2
    }

    private fun ParseHttp(i: Int) {
        ParseHttpUtils.ParseFromHttp(urlPath) { data ->
            if (data != null) {
                getSharedPreferences(packageName, MODE_PRIVATE).edit().putString("url", urlPath).apply()
                this@ShowTextActivity2.data = data
                initViewData(i)
            }
        }
    }

    private fun initViewData(i: Int) {
        if (data!!.isCatalog == 1) {
        } else {
            val book = Book(if (data!!.title == null) "" else data!!.title,
                    Html.fromHtml(if (data!!.text == null) "" else data!!.text).toString())
            flip_view.setBook(book, i)
        }
    }
}
